apps/prairielearn/elements/pl-external-grader-results/pl-external-grader-results.py

```diff
@@ -39,18 +43,18 @@ def ansi_to_html(output):
         return f"[Error converting ANSI to HTML: {e}]\n\n{output}"
 
 
-def prepare(element_html, data):
+def prepare(element_html: str, data: pl.QuestionData) -> None:
     element = lxml.html.fragment_fromstring(element_html)
-    required_attribs = []
-    optional_attribs = []
+    required_attribs: List[str] = []
+    optional_attribs: List[str] = []
     pl.check_attribs(element, required_attribs, optional_attribs)
 
 
-def round_value(val, digits=2):
+def round_value(val: float, digits: int = 2) -> str:
```

Comment:
A little orthogonal to the point of this PR: This function has pretty wonky usage, as it uses two nested calls to string formatting. It would be great if this could be cleaned up some. I think this can be done by using the `round()` function instead of appealing to formatting twice (see https://stackoverflow.com/a/20457115/2923069).

Reply:
I'm a bit worried about `round`'s handling of floating point math; see [here](https://docs.python.org/3/library/functions.html#round). Thoughts?

I think we could get bugs from rounding student's grades down in weird edge cases (e.g., `99.995` becomes `99.99` instead of `100.00`).

Reply:
I think this shouldn't be a big deal, as the value is mainly used for display and the function is currently rounding. I actually think that the current function is calling `round()` under the hood when preparing the floating point number for display. My comment is basically just that we should call this more directly instead of doing two separate string format calls. At the very least, it would be nice to change this to a single format call, instead of two consecutive ones: https://stackoverflow.com/a/1598650